MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -65,19 +65,34 @@ open class SimulatedLocationManager: NavigationLocationManager {
      */
     @objc public init(route: Route) {
         super.init()
+        initializeSimulatedLocationManager(for: route)
+    }
+
+    /**
+     Initalizes a new `SimulatedLocationManager` with the given routeProgress.
+     
+     - parameter routeProgress: The routeProgress of the current route.
+     - returns: A `SimulatedLocationManager`
+     */
+    @objc public init(routeProgress: RouteProgress) {
+        super.init()
+        self.currentDistance = routeProgress.distanceTraveled
+        initializeSimulatedLocationManager(for: routeProgress.route)
+    }
+
+    private func initializeSimulatedLocationManager(for route: Route) {
         self.route = route
-        reset()
         NotificationCenter.default.addObserver(self, selector: #selector(didReroute(_:)), name: .routeControllerDidReroute, object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(progressDidChange(_:)), name: .routeControllerProgressDidChange, object: nil)
     }
     
-    private func reset() {
+    private func reset(_ distance: CLLocationDistance) {
         if let coordinates = route?.coordinates {
             routeLine = coordinates
             locations = coordinates.simulatedLocationsWithTurnPenalties()
             
-            currentDistance = 0
-            currentSpeed = 30
+            currentSpeed = 0
+            currentDistance = distance + (currentSpeed * speedMultiplier)
```

Comment:
I think we should add a new function here for these changes. Also, the units here do not match. We're adding a distance (meters) to a speed (meters per second).

Reply:
Ok

Reply:
@bsudekum this is an already existing function. Remember we had a prior discussion: https://github.com/mapbox/mapbox-navigation-ios/pull/1218#discussion_r179231238

Should we probably leave some comment in the code for it, just to prevent a confusion in the future.